rpc/js/vite.config.ts

```diff
@@ -0,0 +1,36 @@
+import { defineConfig } from 'vite';
+
+import pkg from './package.json';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  define: {
+    'process.env.NODE_ENV': '"production"',
+    __VERSION__: JSON.stringify(pkg.version),
+  },
+  build: {
+    commonjsOptions: {
+      transformMixedEsModules: true,
+      include: [
+        /exponential-backoff/u,
+        /google-protobuf/u,
+        /@improbable-eng\/grpc-web/u,
+        /gen\//u,
+      ],
+    },
```

Comment:
Can you add a comment explaining why this section is needed?

Reply:
@micheal-parks We copied and pasted this from the TS SDK. Do you have any insight as to why this is necessary?

Reply:
The libraries on the list have poor Vite compatibility and adding this config is necessary to transform them into ES modules. We should add a note that this can be removed when we migrate to protobuf-es or a code generating tool that has good support for ES modules.

Reply:
will defer to @micheal-parks to give more concrete details, but this option was necessary to [drastically reduce the bundle size of the typescript sdk](https://github.com/viamrobotics/viam-typescript-sdk/pull/41). I'm assuming we also need a version of this config in this package since we doing CommonJS > ESM conversion.

At minimum the `include` list is wrong, we can remove `/exponential-backoff/u,` since that package is only used in TS-SDK. I think the rest apply.

Reply:
Oh yeah! Forgot that I did that lol, thanks for finding. Yeah, as far as I recall without transforming to ES modules those libraries are not at all tree shakable.

Reply:
Added a comment that says this: 
```
    // This config is necessary to transform libraries on the list into ES modules.
    // This can be removed if protobuf-es or a code generating tool that has good
    // support for ES modules is used.
```
Should I also include the bundle size reduction comment?

Reply:
I think that's sufficient